src/lib.rs

```diff
@@ -55,8 +55,8 @@ struct Pending {
 }
 
 impl Pending {
-    fn prepare_sub_drain(&mut self) -> (&mut Vec<usize>, &Vec<Epoch>) {
-        (&mut self.sub_ref, &self.epoch)
+    fn drain_sub(&mut self) -> (std::vec::Drain<usize>, &mut [Epoch]) {
```

Comment:
niiiice. Just one nit: make the use of `Drain` consistent with the rest of the code, so have `use std::vec::Drain` at the top (right after `use std::sync...` line).